### PR TITLE
MP/2.sequelize model setup

### PR DESCRIPTION
**Pull request by ShephardMP**
I created both User and Salt models using the samples as a base.

I chose to do a different implementation using MD5 hash + salt, when tested, I managed to put some rows into the postgres database with encrypted passwords.

There are some files that I didn't need to touch, so they were left as they were, although they are not in StandardJS.

I wanted to isolate the encryption logic into a separate file, I could make it work, but I feel I put a dependency on the Salt Model into the encryption logic.

(encryptValue takes a value and the saltModel to retrieve the last salt)

### CODE REVIEW

File: migrations/users.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+'use strict'
```
Comment (abelosorio) [inline migrations/users.js]: Same thing here about naming.

File: migrations/users.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      password: {
+        type: Sequelize.STRING,
+        allowNull: false
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
```
Comment (abelosorio) [inline migrations/users.js]: `updatedAt` can be null. `updatedAt` with null value means that the field has never been updated.

File: config/config.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
```
[the remark this answers is not in the document]
Comment (nahudalla) [reply config/config.json]: I was writing about the personal information too, when I saw this comment. I was writing the folowing:

>You should never commit private passwords to `git`! Look in the documentation for `Sequelize.js`, there is a way to configure it using a `.js` file. In that file, you can use the `dotenv` package to load environment variables with the data you need to configure Sequelize. We don't commit files loaded by `dotenv`, you can commit a sample file so other users of your repo know how to configure it and make it work on their own machine.

[the remark this answers is not in the document]
Comment (abelosorio) [reply config/config.json]: Yes, totally. But I think it's ok for the development environment.

File: migrations/users.js (hunk 1 above)
Comment (ShephardMP) [reply migrations/users.js]: changed to 'create-user.js'

File: migrations/users.js (hunk 2 above)
Comment (ShephardMP) [reply migrations/users.js]: changed to:

`
updatedAt: {
        allowNull: true,
        type: Sequelize.DATE
      }
`

File: config/config.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShephardMP) [reply config/config.json]: I changed the names of the databases to reflect the environment.
I had a problem with `chat-platform` as a name for a role in postgresql, so I got rid of the " - ".

I will read about dotenv, seems like a good suggestion.

File: models/user.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+  })
+
+  User.prototype.passwordMatches = async function (value) {
+    const encryptedValue = await encryptValue(value, this.salt)
+    return (encryptedValue === this.password)
+  }
+
+  User.hashPasswordHook = async function (user) {
+    if (!user.password) {
```
Comment (abelosorio) [inline models/user.js]: Two things: 1) Sequelize has its own method to verify if a value has changed; 2) You're not storing the user's salt.

This is how I do it:
```
  User.hashPasswordHook = async function (user, options) {
    if (!user.password || !user.changed('password')) return user

    user.salt = this.getRandomSalt()
    user.password = this.encryptPassword(user.password, user.salt)
  }
```

Let me know if that helps.

File: models/user.js [hunk 5]
```diff
@@ -26,28 +26,27 @@ export default (sequelize, DataTypes) => {
     },
     salt: {
       type: DataTypes.STRING,
-      allowNull: false
+      allowNull: true
```
Comment (abelosorio) [inline models/user.js]: This should be required.

File: models/user.js [hunk 6]
```diff
@@ -26,28 +26,27 @@ export default (sequelize, DataTypes) => {
...
     }
   })
 
-  User.prototype.passwordMatches = async function (value) {
-    const encryptedValue = await encryptValue(value, this.salt)
-    return (encryptedValue === this.password)
+  User.prototype.passwordMatches = function (password) {
+    const encryptedPassword = User.getEncryptedPassword(password)
+    return (encryptedPassword === this.password)
```
Comment (abelosorio) [inline models/user.js]: No need for parenthesis here.

**Pull request merged**